input/docs/handbook/when-activated/index.md

```diff
@@ -40,188 +39,24 @@ public MyCoolViewModel : ISupportsActivation
-
-    flagbug [10:25 AM] i.e always use `WhenActivated` on WPF, if you're writing
-    `this.WhenAnyValue(x => x.ViewModel.CoolProperty).BindWhatever`
-
-    kentcb [10:28 AM] I guess the question is: is there any reason
-    ​_not_​ to use it (because then we can’t simply recommend it as a
-    default stance)
-
-    flagbug [10:30 AM] I guess it's pretty unnecessary on e.g Android
-
-    kentcb [10:31 AM] unnecessary maybe, but does it cause problems? I think
-    it’s useful to be able to say “this is how you can do things” and have that way
-    consistent across all platforms
-
-    paulcbetts [10:31 AM] It's because on WPF, DependencyProperties leak
-
-    kentcb [10:34 AM] @paulcbetts: are you saying that if WPF didn’t leak,
-    `WhenActivated` simply wouldn’t exist?
-
-    paulcbetts [10:34 AM] Well, it would still be useful in certain
-    circumstances
-
-    ​[10:34] You just wouldn't need to wrap every Bind to it
-
-    kentcb [10:35 AM] but Bind takes a VM, and the VM is yet to be assigned in
-    your view’s ctor
-
-    ​[10:36] surely one of the things `WhenActivated` gives us is a point
-    in time at which we ​_know_​ the VM is assigned?
-
-    paulcbetts [10:37 AM] Bind works like WhenAny, it'll just reapply once the
-    VM gets assigned
-
-    kentcb [10:37 AM] ah, that’s right - the VM is just provided for type info
-
-    paulcbetts [10:39 AM] Yep
-
-    grokys [11:56 AM] 
-    regarding DP leaks, what exactly causes the problem with rxui? is it `AddValueChanged`
-    as in http://sharpfellows.com/post/Memory-Leaks-and-Dependency-Properties ?
-
-    i'd like to know so we can avoid it in perspex
-
-    paulcbetts [12:11 PM] 
-    So the problem is, normally when there was a "ValueChanged" event, and you add a handler to it,
-    the lifetime of the handler is tied to the lifetime of the object
+# Do I need WhenActivated?
 
-    So even if you don't free ValueChanged's handler, if the object goes away, you're fine
+If you do a [WhenAny](../when-any) on anything other than `this`, you **need** to put it inside a `WhenActivated` block. If you just launch a window and then that window goes away when app goes away and you have nothing else to manage, you don't need `WhenActivated`. 
 
-    In XAML, if you hook change events on the "Value" property
-    ​_Even when the object goes way_​, you have leaked the event
-    Because it's tied to the static property ValueProperty
+Always use `WhenActivated` with XAML views, if you're writing `this.WhenAnyValue(x => x.ViewModel.Property).Bind()`. You should use it any time there's something your view sets up that will outlive the view - on a Xaml platform, you may have a subscription that you don't want to stay active when the view is detached from the visual tree. It's also useful for setting up things when you get added to the visual tree, although usually the correct place for something like that is in the ViewModel's `WhenActivated`. 
```

Comment:
Don't think Bind() exists as a IObservable, Bindto() does.

this.Bind() is also something you should mention.

Reply:
Could you explain a bit more? I've not quite familiar with code-behind bindings so any help is appreciated :1st_place_medal:

Reply:
so we provide a couple different ways of doing code behind. 

There is a very basic BindTo() method that can hook into any of the WhenAny/Value() type methods. It's very simple and probably not great beyond basic use.

```cs
this.WhenAnyValue(x => x.GreatValue).BindTo(target, target => target.SuperGreatValue);
```

BindTo() is one way so if GreatValue changes, it updates the target.

The other mechanisms in code behind are Bind, OneWayBind etc. 

They use similar mechanisms to get Observables when the target/source objects have changed. So they also suffer from the DependencyProperty issues with the memory leak.

They have some advantages one being simpler notation

```cs
this.Bind(this.ViewModel, vm => vm.GreatValue, view => view.TextBox.GreatValueTextBox); // two way bind
this.OneWayBind(this.ViewModel, vm => vm.GreatValue, view => view.Label.Content); 
```

These bind's also have the ability to convert values back and forth. They can either use lambdas to do this (for once off stuff) or you can register in splat converters that will do it globally. We use this for example to globally convert visibility into bool on WPF for example.